foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/databaseaccess/DatabasePlatform.java

```diff
@@ -3502,46 +3511,69 @@ public void writeAddColumnClause(Writer writer, AbstractSession session, TableDe
-     }
+    /**
+     * INTERNAL:
+     * Returns user name retrieved from JDBC connection.
+     * @throws UnsupportedOperationException on every single call until overridden.
+     */
+    public String getConnectionUserName() {
+        throw new UnsupportedOperationException("Connection user name is not supported.");
+    }
 
-     /**
-      * INTERNAL:
-      * Returns query to check whether given table exists.
-      * Returned query must be completely prepared so it can be just executed by calling code.
-      * @param table database table meta-data
-      * @return query to check whether given table exists
-      */
-     public DataReadQuery getTableExistsQuery(final TableDefinition table) {
-         String column = null;
-         for (FieldDefinition field : table.getFields()) {
-             if (column == null) {
-                 column = field.getName();
-             } else if (field.isPrimaryKey()) {
-                 column = field.getName();
-                 break;
-             }
-         }
-         final String sql = "SELECT " + column + " FROM " + table.getFullName() + " WHERE FALSE";
-         final DataReadQuery query = new DataReadQuery(sql);
-         query.setMaxRows(1);
-         return query;
-     }
+    // Value of shouldCheckResultTableExistsQuery must be false.
+    /**
+     * INTERNAL:
+     * Returns query to check whether given table exists.
+     * Query execution throws an exception when no such table exists.
+     * @param table database table meta-data
+     * @return query to check whether given table exists
+     */
+    protected DataReadQuery getTableExistsQuery(final TableDefinition table) {
+        String column = null;
+        for (FieldDefinition field : table.getFields()) {
+            if (column == null) {
+                column = field.getName();
+            } else if (field.isPrimaryKey()) {
+                column = field.getName();
+                break;
+            }
+        }
+        final String sql = "SELECT " + column + " FROM " + table.getFullName();
+        final DataReadQuery query = new DataReadQuery(sql);
+        query.setMaxRows(1);
+        return query;
+    }
+
+    /**
+     * INTERNAL:
+     * Executes and evaluates query to check whether given table exists.
+     * Query result evaluation depends on {@link #shouldCheckResultTableExistsQuery} value:
+     * Returned value depends on returned vector being non empty when {@code true} and
+     * returned value is always {@code true} when {@code false}.
+     * @param session current database session
+     * @param table database table meta-data
+     * @param suppressLogging whether to suppress logging during query execution
+     * @return value of {@code true} if given table exists or {@code false} otherwise
+     */
+    public boolean checkTableExists(final DatabaseSessionImpl session,
+            final TableDefinition table, final boolean suppressLogging) {
```

Comment:
I see that we are now adding a 'suppressLogging' boolean so that we can pass the value down the inheritance. Is there a usecase in which we would want suppressLogging to be false?
Also, I'm not a fan of breaking it up. I think either we keep the logging switch logic at the `SchemaManager` level, or we move it all (try-finally too) into `DatabasePlatform`

Reply:
suppressLogging false may find some usage in junit tests where you want to see everything.

Reply:
In your unit test, if you'd be testing this impl, then you'd call the Platform classes implementation of `checkTableExists` then, right? Since `SchemaManager's` implementation is literally: 
1) turn off logging, 
2) call platform impl, 
3) turn back on logging.

I personally am not a fan of adding code to the product that is only meant to address testing usage, it clutters things up, but that's just me.
I hope you don't think I'm being a hard-a** about this stuff :/ I know most of this is personal preference.

Reply:
@dazey3, @lukasj: Well, I personally don't much care whether this will be there or not. It's another feature Lukas asked me to add. So you two make some agreement and let me know the result. :)